include/stout/expected.h

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "tl/expected.hpp"
+
+////////////////////////////////////////////////////////////////////////
+
+namespace stout {
+
+////////////////////////////////////////////////////////////////////////
+
+template <typename T, typename Error = std::string>
```

Comment:
Please add a file-level comment explaining what this file does / what it defines.

It's much easier to get an understanding of a file of code when there's an overview comment giving an introduction to why the file exists: https://google.github.io/styleguide/cppguide.html#File_Comments

e.g. in this file we might say:

> "Define an expect class that's used to represent error-or-value types returned by code throughout stout. Errors are `std::string`s, and the error-or-value type is similar to C++23's `std::expected`."

For most readers, that sentence is going to be a lot easier to read and understand compared to the raw C++ below!